src/main/java/core/basesyntax/service/ShopFileReader.java

```diff
@@ -0,0 +1,8 @@
+package core.basesyntax.service;
+
+import core.basesyntax.dto.Transaction;
+import java.util.List;
+
+public interface ShopFileReader {
+    List<Transaction> readFromFile(String fileName);
+}
```

Comment:
```suggestion
public interface FileReader {
    List<String> readFromFile(String fileName);
}
```

let's create independent services
FileReader will be able to read from any file

then let's create a `Parser` service that will take a String and return a Transaction (or take a List of Strings and return a List of Transactions)
if you want to use Validator in your Parser, pass validator to the ParserImpl constructor

Reply:
created Parser